client/src/app/pages/search/components/SearchMenu.tsx

```diff
@@ -0,0 +1,329 @@
+import React from "react";
+import {
+  Menu,
+  MenuContent,
+  MenuItem,
+  MenuList,
+  Popper,
+  SearchInput,
+} from "@patternfly/react-core";
+import { FILTER_TEXT_CATEGORY_KEY } from "@app/Constants";
+import { Label } from "@patternfly/react-core";
+import { useFetchAdvisories } from "@app/queries/advisories";
+import { HubRequestParams } from "@app/api/models";
+import { useFetchPackages } from "@app/queries/packages";
+import { useFetchSBOMs } from "@app/queries/sboms";
+import { useFetchVulnerabilities } from "@app/queries/vulnerabilities";
+
+export interface IEntity {
+  id: string;
+  title: string | null;
+  description?: string | null;
+  navLink: string;
+  type: string;
+  typeColor:
+    | "purple"
+    | "orange"
+    | "blue"
+    | "cyan"
+    | "green"
+    | "red"
+    | "grey"
+    | "gold"
+    | undefined;
+}
+
+// Filter function
+export function filterEntityListByValue(list: IEntity[], searchString: string) {
+  // When the value of the search input changes, build a list of no more than 10 autocomplete options.
+  // Options which start with the search input value are listed first, followed by options which contain
+  // the search input value.
+  let options: React.JSX.Element[] = list
+    .filter(
+      (option) =>
+        option.id.toLowerCase().startsWith(searchString.toLowerCase()) ||
+        option.title?.toLowerCase().startsWith(searchString.toLowerCase()) ||
+        option.description?.toLowerCase().startsWith(searchString.toLowerCase())
+    )
+    .map((option) => (
+      <MenuItem
+        itemId={option.id}
+        key={option.id}
+        description={option.description}
+        to={option.navLink}
```

Comment:
I discovered the reason of the delay while redirecting to pages. the `to` element is causing a "Full page reload". This should be replaced by the native react `useNavigate` or a `<Link/>` component.

Reply:
For me it still shows a delay with either one (a bit improved though), but I will add it and we can always open an issue if you are still seeing it happening.